designs/2019-worker-threads/README.md

```diff
@@ -62,6 +63,17 @@ The `concurrency` option corresponds to `--concurrency` CLI option. Defaults to
 
 This RFC doesn't change `CLIEngine` class because this requires asynchronous API to expose.
 
+### [STEP 1] New `disallowWorkerThreads` top-level property of plugins
+
+This RFC adds `disallowWorkerThreads` top-level property to plugins.
+
+If a loaded plugin has this property with `true`,
+
+- When the `concurrency` option was `"auto"`, ESLint enforces the estimated value to `1`.
+- When the `concurrency` option was `2` or larger, ESLint throws a fatal error.
```

Comment:
Not sure that's a good idea. Users might have a lot of different plugins in their config. Determining which one is causing fatal error is going to be painful. I think if any of the plugins provide `disallowWorkerThreads` property, we should just switch back to sync lining. Maybe we could display a warning, but I see no reason to throw a fatal error.

Reply:
Just the error message can contain the name of caused plugins.

Reply:
Making the existence of any such plugin throw an error means that any plugin that needs to degrade to a non-worker thread setup needs to treat that as a breaking change and release it as a new `major` release.

I see no reason why a simple `warning` wouldn't be enough? It will always work without worker threads, it will just be slower than requested. Warning about it feels to me like the right response then.

Reply:
Good point. That (plugins need a new major release to use this feature) is convincing to me. I will update this section.